containers/hydra/postbuild.py

```diff
@@ -171,26 +173,17 @@ def main():
         print(f'POSTBUILD_INFO="{nixbuildinfo}"')
         print(f'POSTBUILD_INFO_SIGNATURE="{create_signature(nixbuildinfo)}"')
 
-        # generate provenance
-        result = subprocess.run(
-            [
-                "/setup/provenance.sh",
-                niximglink,  # image path
-                nixbuildinfo,  # buildinfo path
-                tmpdir,  # working dir for provenance and sbom
-                provenancename,  # file to save the provenance to
-            ],
-            stdout=subprocess.PIPE,
-            check=False
-        )
+    # Run provenance generation if script is given
+    if PROVENANCE_SCRIPT is not None:
+        result = subprocess.run(PROVENANCE_SCRIPT, stdout=subprocess.PIPE, check=False)
         if result.returncode != 0:
             perror(
-                f"provenance.sh failed ({result.returncode}):"
+                f"PROVENANCE_SCRIPT failed ({result.returncode}):"
```

Comment:
Perhaps rather f"{PROVENANCE_SCRIPT} failed ({result.returncode}):" here?
(Still valid, when you check os.system result code)